text-mode-core/aug/AugState.ts

```diff
@@ -37,12 +39,15 @@ export interface State {
 }
 
 export interface GraphSettings {
+  product?: Product;
   randomSeed?: string;
   viewport: {
-    xmin: number;
-    ymin: number;
-    xmax: number;
-    ymax: number;
+    xmin?: number;
+    ymin?: number;
+    xmax?: number;
+    ymax?: number;
```

Comment:
Yes, if you get the Aug from a graph state, then it always has xmin, etc. But a synthetic Aug (from Text Mode, for example) wouldn't always have xmin, etc. In theory we could require that Text Mode fills in the defaults, but I'd rather have Desmos do that (e.g. 3d uses -5..5 instead of -10..10; it seems like it doesn't quite work as expected right now, but it's a small loss).